client/jetpack-cloud/sections/partner-portal/issue-license-v2/lib/filter.ts

```diff
@@ -0,0 +1,40 @@
+import { APIProductFamilyProduct } from 'calypso/state/partner-portal/types';
+
+/* check if product matches the search query
+ *
+ * @param product - product to check
+ * @param productSearchQuery - search query to match against
+ *
+ * @returns boolean - true if product matches the search query, false otherwise
+ */
+export const isProductMatch = ( product: APIProductFamilyProduct, productSearchQuery: string ) => {
+	const nameFilter = ( name: string ) =>
+		name.toLowerCase().includes( productSearchQuery.toLowerCase() );
+
+	if ( product.slug.startsWith( 'jetpack-complete' ) ) {
+		return [
+			product.name,
+			'Jetpack VaultPress Backup',
+			'Jetpack Scan',
+			'Jetpack Akismet Anti-spam',
+			'Jetpack VideoPress',
+			'Jetpack Boost',
+			'Jetpack Social',
+			'Jetpack Search',
+			'Jetpack Stats',
+			'Jetpack CRM',
+			'Jetpack Creator',
+		].some( nameFilter );
+	}
+
+	if ( product.slug.startsWith( 'jetpack-security' ) ) {
+		return [
+			product.name,
+			'Jetpack VaultPress Backup',
+			'Jetpack Scan',
+			'Jetpack Akismet Anti-spam',
+		].some( nameFilter );
+	}
```

Comment:
@yashwin, I tried to use the `productsIncluded` attribute from the `SelectorProduct`, but it turned out to be trickier than expected. This attribute is a slug, not the actual product name. Therefore, we need to convert those strings back to `SelectorProduct`. Besides, there is no easy way to compare a translatable string and check if it matches a string. This is because it is not guaranteed to be a simple string.

This makes this operation very expensive for a simple search query filter. I decided to keep this as is and simplify everything.

Reply:
After some discussion, I made some changes to use the `productsIncluded` to get a matchable string.